node/podcontroller.go

```diff
@@ -69,7 +69,13 @@ type PodLifecycleHandler interface {
 // of pod status updates asynchronously.
 type PodNotifier interface {
 	// NotifyPods instructs the notifier to call the passed in function when
-	// the pod status changes.
+	// the pod status changes. It should be called pod mutated with the new
+	// pod status.  It may be a pointer to the provider's version of the pod,
+	// and the pointer will not be dereferenced outside of the synchronous
+	// NotifyPods function.
```

Comment:
How are these two statements related? Also, is there any other scenario where the argument is not a pointer to the provider's version of the Pod?

Reply:
A couple things:
* The pod doesn't need to be cloned (DeepCopy) before being passed to notifyPods
* The pod's podstatus should be updated to reflect the podstatus that you want to use to send to API Server

Basically, the intent is not to have the provider just calling notifypods with the pod that got passed from CreatePod, and then only surface the podstatus in getpodstatus. It's really ugly to be honest, in GetPodStatus, GetPods, and NotifyPods, but it means we don't have to callback to GetPodStatus after NotifyPods and we can avoid a nasty race condition.

Reply:
Perhaps, you can suggest better words?

Reply:
```
The provided pointer to a Pod is guaranteed to be used in a read-only fashion.
```